### ISSUE TITLE
Terraform binary does not work on macOS Monterey

### ISSUE THREAD

**Issue by krish7919** [hashicorp/terraform#29880]
I have recently upgraded to macOS Monterey and cannot run any Terraform v0.10.8 commands. However, later versions of Terraform work fine.

```
$ terraform-0.10.8 version
fatal error: runtime: bsdthread_register error

runtime stack:
runtime.throw(0x3140139, 0x21)
	/usr/local/go/src/runtime/panic.go:605 +0x95 fp=0x7ff7bfeff0f0 sp=0x7ff7bfeff0d0 pc=0x1029f35
runtime.goenvs()
	/usr/local/go/src/runtime/os_darwin.go:108 +0x83 fp=0x7ff7bfeff120 sp=0x7ff7bfeff0f0 pc=0x10277d3
runtime.schedinit()
	/usr/local/go/src/runtime/proc.go:482 +0xa1 fp=0x7ff7bfeff160 sp=0x7ff7bfeff120 pc=0x102c8e1
runtime.rt0_go(0x7ff7bfeff198, 0x2, 0x7ff7bfeff198, 0x0, 0x1000000, 0x2, 0x7ff7bfeff400, 0x7ff7bfeff411, 0x0, 0x7ff7bfeff419, ...)
	/usr/local/go/src/runtime/asm_amd64.s:175 +0x1eb fp=0x7ff7bfeff168 sp=0x7ff7bfeff160 pc=0x1054f1b

$ terraform-0.13.6 version
Terraform v0.13.6

Your version of Terraform is out of date! The latest version
is 1.0.10. You can update by downloading from https://www.terraform.io/downloads.html
```

I know this is a pretty old version of Terraform, however, we have legacy projects still using the older version (we are in the process to migrate them to Terraform v1.0) and absolutely require this fix.

Would there be any support provided here to resolve this?

**Comment by krish7919** [comment]
This seems to be a golang issue, would it be possible to build v0.10.8 with a newer version of Go and release?

**Comment by jbardin** [comment]
Hi @krish7919,

Yes, this is going to be a Go issue rather than a Terraform issue. Since the dependencies were entirely vendored there is a good chance that the v0.10 codebase will build with a current Go version in GOPATH mode, but you will have to do some testing to see if there are any unexpected issues.

We use GitHub issues for tracking bugs and enhancements, rather than for questions. While we can sometimes help with certain simple problems here, it's better to use [the community forum](https://discuss.hashicorp.com/c/terraform-core) where there are more people ready to help.

Thanks!

**Comment by krish7919** [comment]
I have tried building it with no success..

**Comment by krish7919** [comment]
Rebuild fixed it, although now we have to rebuild all the providers too. :(

**Comment by krish7919** [comment]
EDIT:

This has been fixed finally by rebuilding Terraform from v0.10.8 and also our custom providers with Go v1.17; thanks for the quick response @jbardin. Although Github issue might not be the place to bring this up, I think many many more devs using pre Go v1.16 Terrafom binaries will hit this issue once Monterey is installed, and will come here looking for solutions.

My thought process was to save the trouble and ask if you could handle it at your end since you might know how many folks in the field are still using pre-Go v1.16 Terraform. ;)
